cartographer_grpc/framework/server.h

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2017 The Cartographer Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CARTOGRAPHER_GRPC_FRAMEWORK_SERVER_H
+#define CARTOGRAPHER_GRPC_FRAMEWORK_SERVER_H
+
+#include <grpc++/grpc++.h>
+#include <cstddef>
+#include <memory>
+#include <string>
+#include <thread>
+
+#include "cartographer_grpc/framework/rpc_handler.h"
+#include "cartographer_grpc/framework/service.h"
+
+namespace cartographer_grpc {
+namespace framework {
+
+class Connection;
+class Server {
+ private:
+  // All options that configure server behaviour such as number of threads,
+  // ports etc.
+  struct Options {
+    size_t number_of_threads = 4;
+    std::string server_address = "0.0.0.0:50051";
+  };
+
+ public:
+  // This 'Builder' is the only way to construct a 'Server'.
+  class Builder {
+   public:
+    std::unique_ptr<Server> Build();
+    void SetNumberOfThreads(size_t number_of_threads);
+    void SetServerAddress(const std::string& server_address);
+
+    template <typename RpcHandler, typename Service>
```

Comment:
I think the names Service and RpcHandler should only be used for classes.
Replace by "ServiceType" etc.

Reply:
Done.